### PR TITLE
Replace use of rowsums in tidyr episode

### LINKED ISSUE #531: Replace use of rowSums() to create summary columns in tidyr episode
### What is the problem?

The use of rowSums() does not fit well into the tidyverse way of doing things. In addition, the way that `interviews_plotting` is currently created counts the `no_listed_items` column, and thus no respondents have 0 items as they should.

Alternative way is to do a grouped mutate to count the rows per `key_ID` after `separate_longer_delim()`. Also need to use `if_else()` to fill in 0 for `"no_listed_items"` and `"none"`.

To-do
- [x] Fix code and explain in tidyr episode
- [x] Change code to create data in ggplot2 episode
- [x] Fix plotting data used in ggplot2 episode

### Location of problem
https://datacarpentry.org/r-socialsci/instructor/04-tidyr.html

### PR DESCRIPTION
**Pull request by jessesadler**
Close #531 

- Use grouped mutate() instead of rowSums() to create summary columns. This is more in line with tidyverse usage and fixes a mistake in lesson where no household was shown to have 0 items.
- Explain use of grouped mutate() and dplyr `if_else()` function.
- Create new exercise to do the same with months lacking food.
- Remove exercise in pivot longer that replicated earlier calculations in a confusing manner.
- Change code to recreate `interviews_plotting` in ggplot2 episode using new method.
- Change `interviews_plotting` data using new method.

### CODE REVIEW

File: episodes/04-tidyr.Rmd [hunk 1]
```diff
@@ -329,10 +390,10 @@ We created some summary tables on `interviews_items_owned` using `count` and
 `summarise`. We can create the same tables on `interviews_long`, but this will
 require a different process.
 
-1. Make a table showing showing the number of respondents in each village who
-  owned a particular item, and include all items. The difference between this
-  format and the wide format is that you can now `count` all the items using the
-  `items_owned` variable.
+Make a table showing showing the number of respondents in each village who owned
```
Comment (juanfung) [inline episodes/04-tidyr.Rmd:393]: `showing` appears twice

**Review by juanfung — CHANGES_REQUESTED** [review]
Looks great! Approve contingent on fixing the one minor typo (left a comment on line 393 of `04-tidyr.Rmd`.

File: episodes/04-tidyr.Rmd (hunk 1 above)
Comment (jessesadler) [reply episodes/04-tidyr.Rmd]: Fixed! Good catch.

**Review by juanfung — APPROVED** [review]
Not clear if this is approving the update but...approve!

**Pull request merged**